keras/wrappers/scikit_learn.py

```diff
@@ -149,8 +179,108 @@ class KerasClassifier(BaseWrapper):
     loss : string
         Loss function used by the model during compilation/training.
     """
-    def __init__(self, model, optimizer='adam', loss='categorical_crossentropy', **kwargs):
-        super(KerasClassifier, self).__init__(model, optimizer, loss, **kwargs)
+
+    def __init__(self, 
+                 model, 
+                 optimizer='adam', 
+                 loss='categorical_crossentropy', 
+                 train_batch_size=128, 
+                 test_batch_size=128,
+                 nb_epoch=100, 
+                 shuffle=True, 
+                 show_accuracy=False,
+                 validation_split=0, 
+                 validation_data=None, 
+                 callbacks=None,
+                 verbose=0, 
+                 class_weight=None):
+        super(KerasClassifier, self).__init__(model, 
+                                              optimizer, 
+                                              loss, 
+                                              train_batch_size, 
+                                              test_batch_size,
+                                              nb_epoch, 
+                                              shuffle, 
+                                              show_accuracy,
+                                              validation_split, 
+                                              validation_data, 
+                                              callbacks,
+                                              verbose)
+
+        self.class_weight = class_weight
+
+
+    def __ohe_output(function):
```

Comment:
I disagree with you on this one. Even if this code was showing only once, I would still be happy yo put it in a decorator. This code has nothing to do with the logic of the fit or score methods. It rather dresses the input of those functions. From my point of view this decorator removes unnecessary code from within the fit and score functions, and therefore makes them more readable. It separates the code into distinct logical parts.